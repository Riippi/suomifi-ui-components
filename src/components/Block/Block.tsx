import React, { Component } from 'react';
import classnames from 'classnames';
import { HtmlDiv, HtmlDivProps } from '../../reset/HtmlDiv/HtmlDiv';

export interface BlockProps extends HtmlDivProps {
  /**
   * Change block semantics
   * @default default
   */
  variant?: 'default' | 'section' | 'header' | 'nav' | 'main' | 'footer';
  className?: string;
}

const baseClassName = 'fi-block';

export class Block extends Component<BlockProps> {
  render() {
    const { className, variant, ...passProps } = this.props;
    const ComponentVariant =
      !variant || variant === 'default' ? HtmlDiv : variant;
    return (
      <ComponentVariant
        {...passProps}
        className={classnames(baseClassName, className)}
      />
    );
  }
}
