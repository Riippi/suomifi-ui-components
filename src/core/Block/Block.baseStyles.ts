import { css } from 'styled-components';
import { BlockProps } from './Block';
import { withSuomifiTheme, TokensAndTheme } from '../theme';
import { element, font } from '../theme/reset';
import { spacingModifiers } from '../theme/utils';

export const baseStyles = withSuomifiTheme(
  ({ theme }: BlockProps & TokensAndTheme) => css`
  ${element({ theme })}
  ${font({ theme })('bodyText')}
  ${spacingModifiers({ theme })('margin')('&.fi-block--margin')}
  ${spacingModifiers({ theme })('padding')('&.fi-block--padding')}
`,
);
