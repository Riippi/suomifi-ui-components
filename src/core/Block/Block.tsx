import React, { Component } from 'react';
import { default as styled } from 'styled-components';
import { withSuomifiDefaultProps } from '../theme/utils';
import { TokensProp, InternalTokensProp } from '../theme';
import {
  Block as CompBlock,
  BlockProps as CompBlockProps,
} from '../../components/Block/Block';
import { SpacingProp } from '../theme/spacing';
import { baseStyles } from './Block.baseStyles';
import classnames from 'classnames';

const baseClassName = 'fi-block';

export interface BlockProps extends CompBlockProps, TokensProp {
  /** Padding from theme */
  padding?: SpacingProp;
  /** Margin from theme */
  margin?: SpacingProp;
}

const StyledBlock = styled(
  ({
    tokens,
    className,
    padding,
    margin,
    ...passProps
  }: BlockProps & InternalTokensProp) => (
    <CompBlock
      {...passProps}
      className={classnames(className, {
        [`${baseClassName}--padding-${padding}`]: !!padding,
        [`${baseClassName}--margin-${margin}`]: !!margin,
        [`${baseClassName}--${passProps.variant}`]: !!passProps.variant,
      })}
    />
  ),
)`
  ${props => baseStyles(props)};
`;

/**
 * Used displaying Block with correct styles
 */
export class Block extends Component<BlockProps> {
  static section = (props: BlockProps) => (
    <StyledBlock {...withSuomifiDefaultProps(props)} variant="section" />
  );

  static header = (props: BlockProps) => (
    <StyledBlock {...withSuomifiDefaultProps(props)} variant="header" />
  );

  static nav = (props: BlockProps) => (
    <StyledBlock {...withSuomifiDefaultProps(props)} variant="nav" />
  );

  static main = (props: BlockProps) => (
    <StyledBlock {...withSuomifiDefaultProps(props)} variant="main" />
  );

  static footer = (props: BlockProps) => (
    <StyledBlock {...withSuomifiDefaultProps(props)} variant="footer" />
  );

  render() {
    return <StyledBlock {...withSuomifiDefaultProps(this.props)} />;
  }
}
