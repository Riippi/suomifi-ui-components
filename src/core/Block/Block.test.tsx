import React from 'react';
import { render } from 'react-testing-library';
import { axeTest } from '../../utils/test/axe';

import { Block } from './Block';

const TestBlock = (
  <div data-testid="test-block">
    <Block>Hey this is test</Block>
    <Block padding="xl">Hey this is test lead</Block>
    <Block.section>Hey this is test bold</Block.section>
  </div>
);

test('calling render with the same component on the same container does not remount', () => {
  const BlockRendered = render(TestBlock);
  const { container } = BlockRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(TestBlock));
