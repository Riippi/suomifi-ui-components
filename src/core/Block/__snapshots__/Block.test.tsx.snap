// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-block--margin-xxs {
  margin: 5px;
}

.c1.fi-block--margin-xs {
  margin: 10px;
}

.c1.fi-block--margin-s {
  margin: 15px;
}

.c1.fi-block--margin-m {
  margin: 20px;
}

.c1.fi-block--margin-l {
  margin: 25px;
}

.c1.fi-block--margin-xl {
  margin: 30px;
}

.c1.fi-block--margin-xxl {
  margin: 40px;
}

.c1.fi-block--margin-xxxl {
  margin: 60px;
}

.c1.fi-block--margin-xxxxl {
  margin: 80px;
}

.c1.fi-block--margin-insetXxs {
  margin: 2px;
}

.c1.fi-block--margin-insetXs {
  margin: 4px;
}

.c1.fi-block--margin-insetS {
  margin: 6px;
}

.c1.fi-block--margin-insetM {
  margin: 8px;
}

.c1.fi-block--margin-insetL {
  margin: 10px;
}

.c1.fi-block--margin-insetXl {
  margin: 16px;
}

.c1.fi-block--margin-insetXxl {
  margin: 20px;
}

.c1.fi-block--padding-xxs {
  padding: 5px;
}

.c1.fi-block--padding-xs {
  padding: 10px;
}

.c1.fi-block--padding-s {
  padding: 15px;
}

.c1.fi-block--padding-m {
  padding: 20px;
}

.c1.fi-block--padding-l {
  padding: 25px;
}

.c1.fi-block--padding-xl {
  padding: 30px;
}

.c1.fi-block--padding-xxl {
  padding: 40px;
}

.c1.fi-block--padding-xxxl {
  padding: 60px;
}

.c1.fi-block--padding-xxxxl {
  padding: 80px;
}

.c1.fi-block--padding-insetXxs {
  padding: 2px;
}

.c1.fi-block--padding-insetXs {
  padding: 4px;
}

.c1.fi-block--padding-insetS {
  padding: 6px;
}

.c1.fi-block--padding-insetM {
  padding: 8px;
}

.c1.fi-block--padding-insetL {
  padding: 10px;
}

.c1.fi-block--padding-insetXl {
  padding: 16px;
}

.c1.fi-block--padding-insetXxl {
  padding: 20px;
}

<div
  data-testid="test-block"
>
  <div
    class="c0 fi-block c1"
  >
    Hey this is test
  </div>
  <div
    class="c0 fi-block c1 fi-block--padding-xxxl"
  >
    Hey this is test lead
  </div>
  <section
    class="fi-block c1 fi-block--section"
  >
    Hey this is test bold
  </section>
</div>
`;
